relay-server/src/utils/sleep_handle.rs

```diff
@@ -0,0 +1,35 @@
+use std::future::Future;
+use std::pin::Pin;
+use std::task::Poll;
+use std::time::Duration;
+
+pub struct SleepHandle(Option<Pin<Box<tokio::time::Sleep>>>);
```

Comment:
Ideally add some doc comments explaining its purpose and behavior, maybe even a minimal example.